distribution/lib/Standard/Base/0.0.0-dev/src/Data/Text/Extensions.enso

```diff
@@ -599,12 +599,41 @@ Text.is_empty = this == ""
 Text.not_empty : Boolean
 Text.not_empty = this.is_empty.not
 
+## Inserts text value at the specified index.
+
+   Arguments:
+     - index: The location in the text to inset text at (0-based). The
+       index is also allowed to be negative, then the characters are
+       counted from the end of the text, i.e. -1 will insert the text after
+       the last x.
+
+   ! What is a Character?
+     A character is defined as an Extended Grapheme Cluster, see Unicode
+     Standard Annex 29. This is the smallest unit that still has semantic
+     meaning in most text-processing applications.
+
+   > Example
+    Insert text at a specified index
+
+         "Hello World!".insert 5 " Cruel" == "Hello Cruel World!"
+         "Hello World!".insert -1 " Cruel" == "Hello World! Cruel"
```

Comment:
May be worth adding an example showing how to insert at the very beginning of the string, i.e. `insert 0`.